Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/IndirectTwoPeakFit.py

```diff
@@ -0,0 +1,391 @@
+
+        self._delete_workspace(parameter_table_name)
+        return parameter_values
+
+    def _fit_workspace(self, input_name, function, output_name, parameters_only):
+        fit_alg = self.createChildAlgorithm("Fit", enableLogging=False)
+        fit_alg.setAlwaysStoreInADS(True)
+        fit_alg.setProperty("Function", function)
+        fit_alg.setProperty("InputWorkspace", input_name)
+        fit_alg.setProperty("Output", output_name)
+        fit_alg.setProperty("OutputParametersOnly", parameters_only)
+        fit_alg.execute()
+
+        self._delete_workspace('__peak_NormalisedCovarianceMatrix')
+
+    def _plot_peak_by_log_value(self, input_string, function, start_x, end_x, fit_type, minimizer, max_iterations,
+                                output_name, create_output=True, composite_members=True, convolve_members=True):
+        plot_alg = self.createChildAlgorithm("PlotPeakByLogValue", enableLogging=False)
+        plot_alg.setProperty("Input", input_string)
+        plot_alg.setProperty("Function", function)
+        plot_alg.setProperty("StartX", start_x)
+        plot_alg.setProperty("EndX", end_x)
+        plot_alg.setProperty("FitType", fit_type)
+        plot_alg.setProperty("Minimizer", minimizer)
+        plot_alg.setProperty("MaxIterations", max_iterations)
+        plot_alg.setProperty("CreateOutput", create_output)
+        plot_alg.setProperty("OutputCompositeMembers", composite_members)
+        plot_alg.setProperty("ConvolveMembers", convolve_members)
+        plot_alg.setProperty("OutputWorkspace", output_name)
+        plot_alg.execute()
+        mtd.addOrReplace(output_name, plot_alg.getProperty("OutputWorkspace").value)
+
+    def _process_indirect_fit_parameters(self, input_name, column_x, x_unit, parameter_names, output_name):
+        pifp_alg = self.createChildAlgorithm("ProcessIndirectFitParameters", enableLogging=False)
+        pifp_alg.setProperty("InputWorkspace", input_name)
+        pifp_alg.setProperty("ColumnX", column_x)
+        pifp_alg.setProperty("XAxisUnit", x_unit)
+        pifp_alg.setProperty("ParameterNames", parameter_names)
+        pifp_alg.setProperty("OutputWorkspace", output_name)
+        pifp_alg.execute()
+        self._result_ws = pifp_alg.getProperty("OutputWorkspace").value
+        mtd.addOrReplace(self._result_name, pifp_alg.getProperty("OutputWorkspace").value)
+
+    def _convert_to_matrix_workspace(self, table_name, column_x, column_y, output_name):
+        ctmw_alg = self.createChildAlgorithm("ConvertTableToMatrixWorkspace", enableLogging=False)
+        ctmw_alg.setProperty("InputWorkspace", table_name)
+        ctmw_alg.setProperty("ColumnX", column_x)
+        ctmw_alg.setProperty("ColumnY", column_y)
+        ctmw_alg.setProperty("OutputWorkspace", output_name)
+        ctmw_alg.execute()
+        mtd.addOrReplace(output_name, ctmw_alg.getProperty("OutputWorkspace").value)
+
+    def _convert_to_histogram(self, input_name, output_name):
```

Comment:
this is only place it is used. So we could change the function def to just take one variable.